api/protogen/protogen.go

```diff
@@ -267,15 +260,60 @@ func terraformAttributeTypeToProtoType(attrName string, attrType attr.Type) (rep
 	case basetypes.StringType:
 		return false, "string", nil, nil
 	case types.ListType:
-		return terraformRepeatedAttributeTypeToProtoType(attrName, v.ElementType())
+		toProtoType, s, m, err := terraformRepeatedAttributeTypeToProtoType(attrName, v.ElementType())
+		if err != nil {
+			return false, "", nil, err
+		}
+
+		if m != nil {
+			m.Name += "Instance"
+			s += "Instance"
+		}
+
+		return toProtoType, s, m, nil
 	case types.SetType:
 		return terraformRepeatedAttributeTypeToProtoType(attrName, v.ElementType())
-	// TODO: Add support for nested objects.
+	case types.ObjectType:
+		return terraformObjectAttributeTypeToProtoType(attrName, v.AttrTypes)
+
 	default:
 		return false, "", nil, fmt.Errorf("unsupported Terraform type: %s", attrType.String())
 	}
 }
 
+// Converts a Terraform object attribute to a Protocol Buffer message type.
+func terraformObjectAttributeTypeToProtoType(attrName string, attrTypes map[string]attr.Type) (repeated bool, protoType string, messages *message, err error) {
+	messageName := schema.ProtoMessageName(attrName)
+	newMessage := &message{
+		Name:   messageName,
+		Fields: []field{},
+	}
+
+	for name, attrType := range attrTypes {
```

Comment:
The order of items when iterating over a map is deterministic.
You must make sure that the behavior of this function is always deterministic.